### PR TITLE
Dice parse

### PR DESCRIPTION
**Pull request by C-Hipple**
Created a parser for #d# to roll a certain number of dice of a particular size.  If you say 3d6 it will roll 3 6-sided dice and tell you the total result and the individual rolls.  If you only roll one die, which can be called either by 1d6 or just d6, it returns just the integer result, allowing it to be backwards compatible with the "bahp" and "rolloff" triggers.

### CODE REVIEW

**Comment by C-Hipple** [comment]
whoops.  Thought it was passing build.  Currently failing on linter.

**Comment by C-Hipple** [comment]
It's passing now.  Fixed some merge issues and linter issues.  Took a few extra commits since nix was out of commission.

File: dice.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import random
+import re
 
 def rollin():
```
Comment (bhipple) [inline dice.py]: Let's change this to:
```
def rollin(diceSize):
   return random.randint(1, diceSize)
```

File: dice.py [hunk 2]
```diff
@@ -7,6 +8,36 @@ def rollOff(text):
     '''Does a rolloff for the listed bros'''
     rolls = []
     for name in text.split()[1:]:
-        rolls += [(rollin(), name)]
+        rolls += [(rollDice("d20"), name)]
```
Comment (bhipple) [inline dice.py]: Has to call `rollin(20)` to just get the result back; the rollDice function will run the string through a parser and will return more string metadata than just the roll result.

File: dice.py [hunk 3]
```diff
@@ -7,6 +8,36 @@ def rollOff(text):
...
+
+    split = match[0].split("d")
+    # If there's a number before the d, it reads it, else sets to 1
+    try:
+        numDice = int(split[0])
+    except:
+        numDice = 1
+    diceSize = int(split[1])
+
+    rolls = [random.randint(1, diceSize) for i in range(numDice)]
```
Comment (bhipple) [inline dice.py]: `rolls = [rollin(diceSize) for i in range(numDice)]`

File: test_brobot.py [hunk 4]
```diff
@@ -80,6 +87,23 @@ def test_d20_priority(self):
...
+        x = runHandlers("Drop 3d6!")
+        x = x.split("\r\n")
+        self.assertTrue(len(x) == 2)
+
+        # Test that it still realizes this is a single dice
+        x = runHandlers("Drop 1d6!")
+        i = int(x)
+        self.assertTrue(i >= 1 and i <= 6)
+
+    def test_badDiceCalls(self):
```
Comment (bhipple) [inline test_brobot.py]: Nice tests!

**Comment by bhipple** [comment]
`default.nix` works just fine on NixOS, but it has issues on other systems.  It appears something is in scope or defaulted on NixOS that behaves differently elsewhere; I'll try to find it.

**Comment by C-Hipple** [comment]
Updated the changes requested it's passing test.  woop woop

**Comment by bhipple** [comment]
You have to run `git pull origin master` and resolve the conflicts with `default.nix`.

**Pull request merged**